### PR TITLE
ONXP-761 [Account settings] Add QR code to Account Settings page(refe…

### PR DESCRIPTION
**Pull request by OlehHinko**
…rral section)

### CODE REVIEW

**Comment by OlehHinko** [comment]
@goonerDp please check the pull request

File: src/assets/styles/_ant-override.scss [hunk 1]
```diff
@@ -28,6 +28,21 @@
...
+		}
+	}
+}
+
+.modal-qrcode {
+	.ant-modal-body {
+		text-align: center;
+	}
+}
+
```
Comment (goonerDp) [inline src/assets/styles/_ant-override.scss]: Move these styles to corresponding scss files.
modals.scss, ...

File: src/components/modals/ShowQRCode.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+					className="modal-qrcode"
+				>
+					<QRCode size={256} value={link} />
+				</Modal>
+			</>
+		);
+	}
+}
+
+export default ShowQRCode;
```
Comment (goonerDp) [inline src/components/modals/ShowQRCode.js]: Why do you use a class component? The simple function would be enough.

File: src/pages/profile/ReferralLink.js [hunk 3]
```diff
@@ -1,32 +1,60 @@
...
-					suffix={
-						<CopyToClipboard
-							text={link}
-							onCopy={() => message.info("Referral link already in the clipboard.", 5)}
-						>
-							<Icon type="copy" style={{ marginLeft: 5, width: 16, height: 16 }} />
-						</CopyToClipboard>
-					}
+const modals = {
+	ADD_SHOW_QR_CODE: "ADD_SHOW_QR_CODE",
```
Comment (goonerDp) [inline src/pages/profile/ReferralLink.js]: ADD_SHOW_QR_CODE => SHOW_QR_CODE_MODAL

File: src/pages/profile/ReferralLink.js [hunk 4]
```diff
@@ -1,32 +1,60 @@
...
+					link={link}
+					isModalVisible={ADD_SHOW_QR_CODE}
+					hideModal={this.hideModalQRCode(modals.ADD_SHOW_QR_CODE)}
 				/>
-			</p>
-		</div>
-	);
+			</div>
+		);
+	}
```
Comment (goonerDp) [inline src/pages/profile/ReferralLink.js]: If you need state in function component use "useState" hook.
https://reactjs.org/docs/hooks-state.html

File: src/pages/profile/ReferralLink.js [hunk 5]
```diff
@@ -1,60 +1,51 @@
...
-
-export default class ReferralLink extends Component {
-	state = {
-		ADD_SHOW_QR_CODE: false,
-	};
-
-	showModalQRCode = type => () => {
-		this.setState({ ADD_SHOW_QR_CODE: true });
+export default function ReferralLink() {
+	const [SHOW_QR_CODE_MODAL, setToggleQRCode] = useState(false);
```
Comment (goonerDp) [inline src/pages/profile/ReferralLink.js]: In this compnent there is only one modal, so its save to name:
SHOW_QR_CODE_MODAL => isModalVisible
setToggleQRCode => showModal

Please, rename.

File: src/pages/profile/ReferralLink.js [hunk 6]
```diff
@@ -1,60 +1,51 @@
...
+	const showModalQRCode = () => {
+		setToggleQRCode(true);
 	};
-
-	hideModalQRCode = type => () => {
-		this.setState({ ADD_SHOW_QR_CODE: false });
+	const hideModalQRCode = () => {
+		setToggleQRCode(false);
```
Comment (goonerDp) [inline src/pages/profile/ReferralLink.js]: `showModalQRCode `and `hideModalQRCode `are unnecessary. You can pas inline function to Button's onClick handler.

**Pull request merged**